processor/resourcedetectionprocessor/factory_test.go

```diff
@@ -51,7 +52,12 @@ func TestInvalidConfig(t *testing.T) {
 	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig()
 	oCfg := cfg.(*Config)
-	oCfg.Detectors = []string{"not-existing"}
+	oCfg.Detectors = []string{"not-existing", "system"}
+	oCfg.DetectorConfig = DetectorConfig{
+		SystemConfig: system.Config{
+			HostnameSources: []string{"dns"},
+		},
+	}
```

Comment:
This looks like unneeded change. Can we discard it?

Reply:
You mean discard `SystemConfig` ?

Reply:
Yes, I have discarded this changes.